src/pngchunk_int.cpp

```diff
@@ -520,9 +520,6 @@ DataBuf PngChunk::readRawProfile(const DataBuf& text, bool iTXt) {
   while ('0' <= *sp && *sp <= '9') {
     // Compute the new length using unsigned long, so that we can check for overflow.
     const size_t newlength = (10 * length) + (*sp - '0');
-    if (newlength > std::numeric_limits<size_t>::max()) {
-      return {};  // Integer overflow.
-    }
```

Comment:
Yes, this overflow check doesn't work. It needs to happen _before_ the calculation of `newlength`! But this whole block of code could probably be replaced with `strtoull` or something similar.